src/lib.rs

```
@@ -430,6 +454,32 @@ impl SecretKey {
         Ok(SecretKey::from_mut(&mut fr))
     }
 
+    /// Deserialize a hex-encoded representation of a `SecretKey` to a `SecretKey` instance.
+    pub fn from_hex(hex: &str) -> Result<Self> {
+        let sk_bytes = hex::decode(hex).map_err(|e| {
+            Error::HexConversionFailed(format!(
+                "Failed to decode hex representation: {}",
+                e.to_string()
+            ))
+        })?;
+        let mut sk_bytes: [u8; SK_SIZE] = sk_bytes.try_into().unwrap_or_else(|v: Vec<u8>| {
+            panic!(
+                "Expected vec of length {} but received vec of length {}",
+                SK_SIZE,
+                v.len()
+            )
```

Same here, no need to panic, we can return an error